compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -5109,6 +5147,21 @@ TR_S390Peephole::perform()
                }
             break;
             }
+
+         case TR::InstOpCode::LHI:
+            {
+            // This optimization is disabled by default because there exist cases in which we cannot determine whether this transformation
+            // is functionally valid or not. The issue resides in the various runtime patching sequences using the LHI instruction as a
+            // runtime patch point for an offset. One concrete example can be found in the virtual dispatch sequence for unresolved calls
+            // on 31-bit platforms where an LHI instruction is used and is patched at runtime.
+            //
+            // TODO: To enable this optimization we need to implement an API which marks instructions that will be patched at runtime and
```

Comment:
Please open a GitHub issue for this TODO and reference it in the code comment.  Otherwise it is likely to be forgotten.